src/utils.c

```diff
@@ -63,6 +62,12 @@ void update_env_script(gchar *prefix, gchar *restraint_url,
     fclose(env_file);
 }
 
+void
+remove_env_file(guint port)
+{
+    gchar *filename = get_envvar_filename(port);
+    g_remove(filename);
```

Comment:
Valgrind tests are failing because filename is not freed before return.

Also note that `Review-checks` should be addressed before requesting reviews. These can be catched before opening a PR running them locally with `make clean all check valgrind`. They should complete successfully before opening the PR, or before inviting reviewers if you are writing a bug fix with a test covering it.

And if review is required with failing tests, just comment in the PR.

Reply:
I am aware Valgrind checks are failing.  I inadvertently ran "make check" instead of "make valgrind" so the intent of doing the right thing was there.  I'm adding another test case which is what's holding up the changeset.

Reply:
Even before valgrind test, I woke up at 3 AM realizing I forgot to free this memory so I'm on top of it.